(feat): Agent Control (#19943)
newrelic/docs-website: src/content/docs/new-relic-control/fleet-control/setup.mdx

## Create configuration versions
After a configuration is created, you can update it by creating a newer version of this configuration. These different versions can be selected when adding them to a fleet. 

To create a new version, select a configuration, then click **Create config version**. This creates a new version of the configuration, which you can update within a fleet.

Versioning plays a crucial role in protecting the state of fleets. 

<Callout variant="important">
  To update a fleet with the new configuration version, you must manually select it, as the fleet does not automatically update to the new version.
</Callout>

## Create a fleet
With Agent Control installed and the configurations ready to apply to each cluster, you’re now ready to create and deploy your fleet.

Follow these steps to create and deploy a fleet:

1. In New Relic, go to **New Relic Control → Fleets**, and then click **Create a fleet**.
2. Name the fleet, select the fleet type, and set the Access management of that fleet.

<Callout variant="important">
  Only **Kubernetes** is supported as the fleet type for the public preview.
</Callout>

> By setting additional access management, users in the selected group can edit the fleet, rather than only Org Product Admins having this capability. Refer to the prerequisites for Fleet Control to learn more about setting up custom roles.

Reviewer: ```suggestion
```

Reviewer: We decided to not include Fine Grained Access for the Public Preview, please remove this step

> <Callout variant="important">

Reviewer: ```suggestion
```

>   You can edit custom roles via [the `customRoleUpdate` API](/docs/apis/nerdgraph/examples/nerdgraph-manage-groups/#update-role).

Reviewer: ```suggestion
  You can edit custom roles via [the `customRoleUpdate` API](/docs/apis/nerdgraph/examples/nerdgraph-manage-groups/#update-role).
```

> </Callout>

Reviewer: ```suggestion
```

## Start deployment
After you create a fleet, click **Start a deployment** to start editing your fleet.

In the panel that slides in from the right:
1. Name your deployment and provide a description. 
2. Next add your configuration(s) by clicking **Add a configuration**.
3. Then add your cluster(s) by clicking **Add managed entities**.<br/>Only clusters that have Agent Control installed and are not already assigned to another fleet appear.

> 

Reviewer: ```suggestion
4. Click **Save draft**, then click **Start deployment**.
```

> 

Reviewer 2: <Callout variant="important">
When you save a draft of your deployment, several validation checks are run against your configuration. Below are some of the most common along with how to resolve them.

- You must have either a infra or NRDOT configuration for any fleet deployment. Agents require a configuration for deployment so you must select one.
- You must not have both infra and NRDOT configurations. Infra and NRDOT configurations are not compatible as they use different protocols. You must pick only one.
- You must have APM operator configuration if you have added any APM agent configurations.

> 

Reviewer 2: Under 4., we should include the following.

**Note:** When clicking on the **Save draft** button a number of configuration validations are run against your configuration. Some of the most common and their resolutions can be found below.

Author: Unclear - can you please elaborate?

<Callout variant="important">

>   You can add only one configuration per agent type to a fleet. Multiple configurations of the same agent type (for example, different configurations of the Kubernetes infrastructure agent) are not supported.

Reviewer: ```suggestion
  You can add only one configuration per agent type to a fleet. Multiple configurations of the same agent type (for example, different configurations of the Kubernetes infrastructure agent) are not supported.
  
  Note an Infrastructure configuration and an OTel configuration cannot be in the same fleet.
  Also note a Fluent Bit agent must be accompanied by either an Infrastructure or OTel configuration.
```

Reviewer 2: We also need to add the following as a catch all:

You must have APM operator configuration if you have added any APM agent configurations

Reviewer 2: ```suggestion
You can add only one configuration per agent type to a fleet. Multiple configurations of the same agent type (for example, different configurations of the Kubernetes infrastructure agent) are not supported.
  
**Note:** An Infrastructure configuration and an NRDOT configuration cannot be in the same fleet.
A Fluent Bit configuration must be accompanied by either an Infrastructure or OTel configuration.
You must have APM operator configuration if you have added any APM agent configurations	  
```

</Callout>

## Deployment draft

> After you add the deployment name, a draft is automatically created for the deployment. Any changes thereafter are saved with this draft in case you navigate away from your current screen.

Reviewer: ```suggestion
After you add the deployment name, add a configuration, or add a managed entity, you will be able to save your draft of your deployment. Be sure to save your draft in case you navigate away from your current screen. Note that you must save your draft before starting the deployment.
```

> To get back to your draft, navigate to **New Relic Control → Fleets**, and open the Deployments tab.

Reviewer 3: ```suggestion
To get back to your draft, navigate to **New Relic Control → Fleets**, and open the **Deployments** tab.
```

> To get back to your draft, navigate to **New Relic Control → Fleets**, and open the **Deployments** tab.

Reviewer: ```suggestion
To get back to your draft, navigate to **New Relic Control → Fleets**, and click the **Deployments** tab.
```

> 

Reviewer: Add this image here:
https://drive.google.com/file/d/18pYQLg5QNbSRWKdTUTCUAqaYdpc4RR8N/view?usp=sharing

Author: I am not sure this image adds any value here. Please let me know if you feel strongly about this.

## Canary
After you've added your configurations and clusters, you can set up a canary as an extra safeguard before deploying.

To do this, click the three dots next to the cluster(s) you want to designate as canary clusters. These clusters will be deployed first, and deployment to non-canary clusters will only proceed once the initial deployments are successful. If any canary deployment fails, the process will halt.

This precaution allows you to test your fleet setup with a select few clusters before deploying to all of them.

## Start a deployment

> When you are ready, click **Start deployment**. You'll see an In progress screen for your deployment. 

Reviewer 3: ```suggestion
When you are ready, click **Start deployment**. You'll see an **In progress** screen for your deployment. 
```

> The In Progress screen shows all of the clusters the configuration is being deployed to. It shows the number of clusters and agents the config is deployed to, as well as failures:

Reviewer 3: ```suggestion
The **In Progress** screen shows all of the clusters the configuration is being deployed to. It shows the number of clusters and agents the config is deployed to, as well as failures:
```

Check the capitalization for In progress...inconsistent between this line and the last.

<Callout variant="important">
  Every configuration in a fleet applies to all clusters within it. Therefore, clusters in a fleet should be homogeneous, as they will all share the same configurations.
</Callout>

> 

Reviewer: Please add this image showing a config applying to a fleet:
https://drive.google.com/file/d/1aart8lA0LjI3g7PvVId4g239JgILu34x/view?usp=sharing

## Additional deployment factors
- **Continuation:** After the initial deployments succeed, the deployment process continues to the remaining clusters in the fleet.
- **Handling failures:** If any subsequent cluster deployment fails, the process continues for the rest of the clusters. The clusters in the initial ring have already validated the configuration, allowing the deployment to proceed to subsequent rings.

> After your deployment completes, your fleet is officially set up! You can now manage all agents within your fleet with a single click, eliminating the need to write scripts for individual agent updates.

Reviewer 3: ```suggestion
After your deployment is complete, your fleet is officially set up! You can now manage all agents within your fleet with a single click, eliminating the need to write scripts for individual agent updates.
```
